### DISCUSSION TITLE
Adding Type Hints

### DISCUSSION THREAD

**Discussion by daikikatsuragawa** [interpretml/DiCE#331] [Ideas]
I propose to add "[Type Hints](https://docs.python.org/3/library/typing.html)" to the code written in Python, which is not in this project. The benefits are as follows.

- Improved code readability
- Improved code robustness (but also requires implementation of other tools such as [mypy](https://github.com/python/mypy))

**Comment by amit-sharma** [comment]
That will be really useful, @daikikatsuragawa You can go ahead and raise a PR for this. mypy is the standard tool to do this, so including mypy (perhaps in the dev/test dependencies) sounds good.

**Reply by daikikatsuragawa** [reply]
@amit-sharma Thanks! I will proceed with the plan.
